auacm/app/modules/submission_manager/judge.py

```diff
@@ -0,0 +1,225 @@
+    '''Compile the submission.'''
+    directory = directory_for_submission(submission)
+    filename = uploaded_file.filename
+    name, ext = filename.rsplit(".", 1)
+    # Don't compile file types that we can't compile.
+    if COMPILE_COMMAND[ext] is None:
+        return COMPILATION_SUCCESS
+    result = subprocess.call(
+        shlex.split(COMPILE_COMMAND[ext].format(os.path.join(directory, name))),
+        stderr=open(os.path.join(directory, "error.txt"), "w")
+    )
+    if result == 0:
+        return COMPILATION_SUCCESS
+    else:
+        submission.update_status('compile')
+        submission.emit_status("compile", -1)
+        return COMPILATION_ERROR
+
+
+def execute_submission(submission, uploaded_file):
+    '''Run the submission.
+
+    TODO(djshuckerow): This method magically got out-of-hand.  Refactor.
+    This method:
+        1. detects all the input files associated with this problem.
+        2. runs the submission with each input file
+        3. checks the performance of the submission for errors (TLE, RTE, etc)
+        4. compares the output against golden test output to verify correctness
+    '''
+    # Initial setup
+    problem = submission.get_problem()
+    problem_directory = directory_for_problem(submission)
+    submission_directory = directory_for_submission(submission)
+    filename = uploaded_file.filename
+    name, ext = filename.rsplit(".", 1)
+    input_path = os.path.join(problem_directory, "in")
+    output_path = os.path.join(problem_directory, "out")
+    # Iterate over all the input files.
+    for fname in os.listdir(input_path):
+        f = os.path.join(input_path, fname)
+        if os.path.isfile(f):
+            # Prepare to run the test file.
+            test_number = int(fname.split(".")[0].strip("in"))
+            out_file = "out{0}.txt".format(test_number)
+            submission.emit_status("running", test_number)
+            max_runtime = problem.time_limit * TIMEOUT_MULTIPLIER[ext]
+            execution = JudgementThread(
+                submission, uploaded_file, f, out_file, max_runtime)
+            start_time = time.time()
+            execution.start()
+            execution.join(max_runtime)
+            # Check the execution for timeouts and runtime errors.
+            if time.time() >= start_time + max_runtime:
+                execution.process.kill()
+                submission.update_status("timeout")
+                submission.emit_status("timeout", test_number)
+                return TIMELIMIT_EXCEEDED
+            elif execution.process.poll() != 0:
+                submission.update_status("runtime")
+                submission.emit_status("runtime", test_number)
+                return RUNTIME_ERROR
+            result_path = os.path.join(submission_directory, "out")
+            # The execution is completed.  Check its correctness.
+            with open(os.path.join(output_path, out_file)) as golden_result, \
```

Comment:
Done.

Also, for future reference, the inspiration is https://en.wikipedia.org/wiki/Gold_standard_(disambiguation)
Golden tests in computing are tests where the results are graded against some ground-truth collection of data, like we have here.  Goldens can be useful for integration tests or "polish" tests -- you can test a frontend's API calls against the backend code with mock data and golden results, or (especially in the world of apps -- it's harder to do this on the web) you can test the appearance of every page to be sure that you don't accidentally break your UI in a section of the app that you never visit.